src/cclib/method/calculationmethod.py

```diff
@@ -31,24 +31,45 @@ class Method(object):
     >>> import cda, cspa, density, fragments, lpa, mbo, mpa, nuclear, opa, population, volume
     """
 
-    def __init__(self, data, progress=None, loglevel=logging.INFO, logname="Log"):
+    def __init__(self, data, requiredAttr, progress=None, loglevel=logging.INFO, logname="Log"):

```

Comment:
I don't think this should be passed in as a parameter. It should default to an empty list, and subclasses of `Method` can override as necessary. See https://github.com/cclib/cclib/blob/master/src/cclib/io/filewriter.py#L31 and https://github.com/cclib/cclib/blob/master/src/cclib/io/wfxwriter.py#L93.